### PR TITLE
[FIX] Global Search checkbox inital state

### PR DESCRIPTION
**Pull request by dpanchuk**
This is fix for search checkbox.
The problem was in giving "global search enabled" option value to "search all" payload by default.
#21659

### CODE REVIEW

**Comment by glebPo** [comment]
👍

**Comment by serjotaylor** [comment]
finally!!!

**Comment by ankar84** [comment]
@ggazzo please take a look at that fix
cc @dudanogueira @debdutdeb

**Comment by ankar84** [comment]
There is another community fix https://github.com/RocketChat/Rocket.Chat/pull/20594

File: app/search/client/provider/result.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ Template.DefaultSearchResultTemplate.onCreated(function () {
 
 	// global search
 	this.globalSearchEnabled = this.data.settings.GlobalSearchEnabled;
-	this.data.parentPayload.searchAll = this.globalSearchEnabled;
+	this.data.parentPayload.searchAll = (!!this.globalSearchEnabled && !!this.data.parentPayload.searchAll);
```
Comment (debdutdeb) [inline app/search/client/provider/result.js:82]:
```suggestion
	this.data.parentPayload.searchAll = false;
```
You can just set it to false

**Review by debdutdeb — CHANGES_REQUESTED** [review]
HI @dpanchuk thanks for this PR!

Initially I was thinking about changing the element to reflect what actually is being done (i.e. keep it checked instead of unchecked if global search is enabled).

BUT, I like the idea of not defaulting to that behaviour but leave it as a toggle for the users to switch manually, who actually is meaning to use it.

**Comment by dudanogueira** [comment]
Hi! Thanks for your contribution. We will close this PR in favor of 
https://github.com/RocketChat/Rocket.Chat/pull/25568

**Pull request closed without merging**